test/performance/core/char_operations/char_predicate_benchmark.cpp

```diff
@@ -26,41 +42,63 @@ static void simple(benchmark::State & state)
     for (auto _ : state)
     {
         i = (i + 1) % (1 << 20);
-        if constexpr (stl)
-            sum += std::isupper(arr[i]);
-        else
-            sum += is_upper(arr[i]);
+        if constexpr (id == tag::std)
+            sum += std::isalpha(arr[i]);
+        else if constexpr (id == tag::seqan3)
+            sum += seqan3::is_alpha(arr[i]);
+#if SEQAN3_HAS_SEQAN2
+        else if constexpr (id == tag::seqan2)
+            sum += seqan::IsAlpha{}(arr[i]);
+#endif
     }
 
     // prevent complete optimisation
     [[maybe_unused]] volatile size_t fin = sum;
 }
 
-BENCHMARK_TEMPLATE(simple, true);
-BENCHMARK_TEMPLATE(simple, false);
+BENCHMARK_TEMPLATE(simple, tag::std);
+BENCHMARK_TEMPLATE(simple, tag::seqan3);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(simple, tag::seqan2);
+#endif
 
-template <bool stl>
+// ============================================================================
+//  combined
+// ============================================================================
+
+template <tag id>
 static void combined(benchmark::State & state)
 {
     size_t sum = 0;
     size_t i = 0;
 
-    [[maybe_unused]] auto constexpr mycheck = is_punct || is_upper || is_digit;
```

Comment:
are you sure that storing a variable has no performance difference compared to using it directly?

Reply:
Yep!